### ISSUE TITLE
Selection not respecting minimum dimensions settings

### ISSUE THREAD

**Issue by hugofelp** [DominicTobias/react-image-crop#206]
When you drag a selection from one corner past the diagonally opposite corner, the selection flips, respecting the set `minWidth` and `minHeight`, but when the selection is closer to the edge of the image than the min dimensions, it still flips, making the selection smaller than the allowed values.

I don't think this affects the described behavior, but just to provide an accurate description: my component also sets `keepSelection=true` and `crop` has `aspect: 1` defined.

Package versions are as follows:
```
"react": "16.4.1",
"react-dom": "16.4.1",
"react-image-crop": "6.0.0",
```

**Comment by DominicTobias** [comment]
Thanks I'll take a look at this but won't be immediately. Will update this when there is a fix.

**Comment by tudorpavel** [comment]
I think I'm experiencing the same issue, but my expectation is to not have the diagonal flip/invert at all if the rectangle is at the edge of the image.

The issue is that this behavior circumvents `keepSelection=true`, meaning you can disable selection by inverting the rectangle on a corner of the image and then having the possibility to start a new selection elsewhere.

Best explained by a GIF:

![react-image-crop-invert](https://user-images.githubusercontent.com/2506871/48141466-a6303280-e2b3-11e8-8fb4-e21d79d7944b.gif)

**Comment by tudorpavel** [comment]
I've played around with the code in `this.crossOverCheck()`, it's not very straightforward to prevent crossover at the edges of the image. I would actually want a prop `preventCrossover` to disable flipping altogether, what do you think?

**Comment by DominicTobias** [comment]
Fixed in 6.0.5

**Comment by DominicTobias** [comment]
* Note I improved something in this behaviour in 6.0.7
